### ISSUE TITLE
Bug: Relation ignored if same name as Collection

### ISSUE THREAD

**Issue by tomh4** [Barelydead/strapi-plugin-populate-deep#38]
I think the latest addition for the ciruclar references introduces a case, where a relationship is ignored while populating if it is named the same as the collection.
E.g:
Collection named "Response" --> strapis plural is responses
The same collection as a field named "responses" which is not the same collection, just a list of values

--> It gets ignored since version 3.0.0

**Comment by tholden92** [comment]
Is this still an issue?

**Comment by amjadniazi48** [comment]
Yes I have posts with many to many relationship with category but it is not showing

**Comment by 20x-dz** [comment]
Yes. And until recently I didn't bother looking into it as we continued using v2.0.0 just fine. But now it breaks with the latest strapi version (v4.24.5).

I'll look into the proposed fix in https://github.com/Barelydead/strapi-plugin-populate-deep/issues/42#issuecomment-1744516873, as that issue seems to be a duplicate of this one. Wish me luck… 🤞 

[edit] Nevermind, we forgot to remove a deleted component from a dynamic zone. staying at v2.0.0 for now. ✌️
